homeassistant/components/auth/indieauth.py

```diff
@@ -76,18 +77,17 @@ def handle_starttag(self, tag, attrs):
 
     We do not implement extracting redirect uris from headers.
     """
-    session = hass.helpers.aiohttp_client.async_get_clientsession()
     parser = LinkTagParser('redirect_uri')
     chunks = 0
     try:
-        resp = await session.get(url, timeout=5)
-
-        async for data in resp.content.iter_chunked(1024):
-            parser.feed(data.decode())
-            chunks += 1
-
-            if chunks == 10:
-                break
+        async with aiohttp.ClientSession() as session:
```

Comment:
This function will not called a lot? I think now it's also possible to floot the client with redirect_uri request until we run out of space as DOS.

Reply:
Yeah, it is a possibility to get a Home Assistant instance to make an outbound request. It's using a new client session and limits itself to 10kB. We've adopted this approach from IndieAuth but it doesn't feel right. I have it on my list to research further protections.